src/overlays/actors/ovl_En_S_Goro/z_en_s_goro.c

```diff
@@ -74,85 +102,1361 @@ static DamageTable D_80BBF858 = {
+    { &gGoronUnrollAnim, -2.0f, 0, -1, ANIMMODE_ONCE, -4 },
+    { &gGoronShiverAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &gGoronShiverAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &object_taisou_Anim_004DD4, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &object_taisou_Anim_00283C, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &object_taisou_Anim_007764, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &object_taisou_Anim_005790, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &gGoronCoverEarsAnim, 1.0f, 0, -1, ANIMMODE_LOOP, 0 },
+    { &object_taisou_Anim_002C48, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &gGoronStandingHandTappingAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -4 },
+    { &gGoronSleepyAnim, 1.0f, 0, -1, ANIMMODE_ONCE, -4 },
+    { &gGoronStandingIdleAnim, 1.0f, 0, -1, ANIMMODE_LOOP, -8 },
+};
+
+static TexturePtr sEyeTextures[] = {
+    gGoronEyeOpenTex,
+    gGoronEyeHalfTex,
+    gGoronEyeClosedTex,
+    gGoronEyeClosed2Tex,
+};
+
+u16 EnSGoro_ShrineGoron_NextTextID(EnSGoro*, PlayState*);
+u16 EnSGoro_BombshopGoron_NextTextID(EnSGoro*, PlayState*);
+
+s32 EnSGoro_SetPartnerActor_GoronElder(EnSGoro*, PlayState*);
+s32 EnSGoro_SetPartnerActor_GoronChild(EnSGoro*, PlayState*);
+s32 EnSGoro_CheckLullaby(EnSGoro*, PlayState*);
+s32 EnSGoro_CheckGKBehavior(EnSGoro*, PlayState*);
+
+void EnSGoro_ActionDefault(EnSGoro*, PlayState*);
+void EnSGoro_ActionWinterShrineGoron(EnSGoro*, PlayState*);
+void EnSGoro_ActionWinterShrineGoronTalking(EnSGoro*, PlayState*);
+void EnSGoro_ActionSpringShrineGoron(EnSGoro*, PlayState*);
+void EnSGoro_ActionSpringShrineGoronTalking(EnSGoro*, PlayState*);
+void EnSGoro_ActionShopGoron(EnSGoro*, PlayState*);
+void EnSGoro_ActionShopGoronFinishUnroll(EnSGoro*, PlayState*);
+void EnSGoro_ActionShopGoronTalking(EnSGoro*, PlayState*);
+void EnSGoro_ActionShopTakePayment(EnSGoro*, PlayState*);
+void EnSGoro_ActionShopFinishTransaction(EnSGoro*, PlayState*);
+void EnSGoro_ActionAsleep(EnSGoro*, PlayState*);
+void EnSGoro_ActionAsleepMessageWait(EnSGoro*, PlayState*);
+
+s32 EnSGoro_UpdateCheerAnimation(EnSGoro*, PlayState*);
+s32 EnSGoro_UpdateRotationToPlayer(EnSGoro*, PlayState*);
+s32 EnSGoro_UpdateAttentionTarget(EnSGoro*, PlayState*);
+void EnSGoro_UpdateToHandtapAnimation(EnSGoro*);
+void EnSGoro_UpdateSleepyAnimation(EnSGoro*);
+void EnSGoro_UpdateToIdleAnimation(EnSGoro*);
+void EnSGoro_UpdateEyes(EnSGoro*);
+void EnSGoro_UpdateActorFocus(EnSGoro*);
+void EnSGoro_UpdateSleeping(EnSGoro*, PlayState*);
+void EnSGoro_UpdateCollider(EnSGoro*, PlayState*);
+
+s32 EnSGoro_CalculateTranslateRotate(s16, s16, Vec3f*, Vec3s*, s32, s32);
+s32 EnSGoro_OverrideLimbDraw(PlayState* play, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx);
+void EnSGoro_TransformLimbDraw(PlayState*, s32, Actor*);
+void EnSGoro_DrawUnrolled(EnSGoro*, PlayState*);
+void EnSGoro_DrawRolledUp(EnSGoro*, PlayState*);
+
+/*
+        Dialogue tree for EnSGoro when use in Goron Shrine context. Returns ID of next message to display.
+*/
```

Comment:
```suggestion
/**
 * Dialogue tree for EnSGoro when use in Goron Shrine context. Returns ID of next message to display.
 */
```
Generally for function comments like this we do `/**` with `*` on each line for Doxygen. I also don't really think there is much need for a bunch of whitespace before the actual comment.